### PR TITLE
Add azure openai as an option in _llm.py

### PR DESCRIPTION
**Pull request by SliverBulle**
## Description

Thanks a lot for sharing this light accomplishment version of graphrag with the support of incremental insert.  

When you have an `AZURE_OPENAI_API_KEY` instead of `OPENAI_API_KEY` like me, you can use this PR to handle it.
Mimicking the author's writeup, I wrote the asynchronous call to azure openai.
This pull request introduces an azure openai option to LLM and embedding, allowing users who have azure openai key instead of openai key.

## Checklist

- [√ ] I have tested these changes locally.
- [ √] I have reviewed the code changes.
- [ ] I have updated the documentation (if necessary).
- [ ] I have added appropriate unit tests (if applicable).

## How to use

demos are like, I use load_dotenv to load environment variables.
```
from nano_graphrag import GraphRAG, QueryParam
from dotenv import load_dotenv

load_dotenv()

graph_func = GraphRAG(working_dir="./dickens")

with open("./book.txt") as f:
    graph_func.insert(f.read())

# Perform global graphrag search
print(graph_func.query("What are the top themes in this story?"))

# Perform local graphrag search (I think is better and more scalable one)
print(graph_func.query("What are the top themes in this story?", param=QueryParam(mode="local")))
```

you can add an .env file like in your demo.py
```
API_KEY_EMB = "<your azure openai key for embedding>"
AZURE_ENDPOINT_EMB = "<your azure openai endpoint for embedding>"
API_VERSION_EMB ="<api version>"

AZURE_OPENAI_API_KEY="<your azure openai key for embedding>"
AZURE_OPENAI_ENDPOINT="<AZURE_OPENAI_ENDPOINT>"
OPENAI_API_VERSION="<OPENAI_API_VERSION>"
```

or run `export API_KEY_EMB = "<your azure openai key for embedding>"` in your terminal.

### CODE REVIEW

File: nano_graphrag/_llm.py [hunk 1]
```diff
@@ -79,3 +80,80 @@ async def openai_embedding(texts: list[str]) -> np.ndarray:
...
+        system_prompt=system_prompt,
+        history_messages=history_messages,
+        **kwargs,
+    )
+
+async def azure_gpt_35_turbo_complete(
+    prompt, system_prompt=None, history_messages=[], **kwargs
+) -> str:
+    return await azure_openai_complete_if_cache(
+        "gpt-35-turbo",
```
Comment (gusye1234) [inline nano_graphrag/_llm.py:136]: Hi, does azure openai support `gpt-4o-mini`? Maybe change the 35 to 4o-mini is better

File: nano_graphrag/_llm.py [hunk 2]
```diff
@@ -79,3 +80,80 @@ async def openai_embedding(texts: list[str]) -> np.ndarray:
...
+)
+async def azure_openai_embedding(texts: list[str], **kwargs) -> np.ndarray:
+    azure_openai_client = AsyncAzureOpenAI(
+        api_key=os.environ.get("API_KEY_EMB"),
+        api_version=os.environ.get("API_VERSION_EMB"),
+        azure_endpoint=os.environ.get("AZURE_ENDPOINT_EMB")
+    )
+    async with azure_openai_client as client:
+        response = await client.embeddings.create(
+            model="text-embedding-3-large", input=texts, **kwargs
```
Comment (gusye1234) [inline nano_graphrag/_llm.py:157]: Maybe change to text-small? To align with the OpenAI version

File: nano_graphrag/_llm.py (hunk 1 above)
Comment (SliverBulle) [reply nano_graphrag/_llm.py:136]: Thanks for review. Yes, azure openai support `gpt-4o-mini`. I'll write a function named `azure_gpt_4o_mini_complete` and set 4o-mini as the default cheap_model_func.

File: nano_graphrag/_llm.py (hunk 2 above)
Comment (SliverBulle) [reply nano_graphrag/_llm.py]: That's a good question, I've changed it in nearest commit.

**Comment by gusye1234** [comment]
LGTM! 
Do you finish your PR? If so, I will some small things and merge this PR

**Comment by SliverBulle** [comment]
> LGTM! Do you finish your PR? If so, I will fix some small things and merge this PR

I think it's finished, thanks for your time.

**Pull request merged**